packages/mobile/src/transactions/send.ts

```diff
@@ -109,9 +115,7 @@ export function* sendTransactionPromises(
     `Going to send a transaction with id ${context.id}`
   )
 
-  const stableToken = yield getTokenContract(CURRENCY_ENUM.DOLLAR)
-  const stableTokenBalance = yield call([stableToken, stableToken.balanceOf], account)
-
+  const stableTokenBalance = new BigNumber(yield select(stableTokenBalanceSelector))
```

Comment:
i like the check the cache-first approach, seems it could only improve the flow. worst case, its not in the case and we fall to the same logic that already exists 👍

Reply:
I spent about 30 minutes trying to figure out how to do this in a way that will actually reduce the number of calls in the common case, and could not think of a clean solution. I've reverted this change.